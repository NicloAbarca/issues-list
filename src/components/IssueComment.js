import React from 'react';
import ReactMarkdown from 'react-markdown';
import UserWithAvatar from './UserWithAvatar';
import { insertMentionLinks } from '../utils/stringUtils';

export default function IssueComment({ comment }) {
  return (
    <div className="issue-detail__comment">
      <UserWithAvatar user={comment.user} orientation="horizontal"/>

      <div className="issue-detail__comment__body">
        <ReactMarkdown className="markdown" source={insertMentionLinks(comment.body)}/>
      </div>
    </div>
  );
}
