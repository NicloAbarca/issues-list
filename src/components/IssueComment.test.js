import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { checkSnapshot } from '../utils/testUtils';
import IssueComment from './IssueComment';

Enzyme.configure({ adapter: new Adapter() });
it('renders a comment', () => {
  const comment = {
    user: {
      login: "somebody",
      avatar_url: "http://foo"
    },
    body: "a comment"
  };
  const tree = shallow(
    <IssueComment comment={comment}/>
  );
  checkSnapshot(tree);
});