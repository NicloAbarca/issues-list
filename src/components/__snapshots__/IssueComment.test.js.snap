// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a comment 1`] = `
<div
  className="issue-detail__comment"
>
  <UserWithAvatar
    orientation="horizontal"
    user={
      Object {
        "avatar_url": "http://foo",
        "login": "somebody",
      }
    }
  />
  <div
    className="issue-detail__comment__body"
  >
    <ReactMarkdown
      astPlugins={Array []}
      className="markdown"
      escapeHtml={true}
      plugins={Array []}
      rawSourcePos={false}
      renderers={Object {}}
      skipHtml={false}
      source="a comment"
      sourcePos={false}
      transformLinkUri={[Function]}
    />
  </div>
</div>
`;

exports[`test renders a comment 1`] = `
<div
  className="issue-detail__comment">
  <UserWithAvatar
    orientation="horizontal"
    user={
      Object {
        "avatar_url": "http://foo",
        "login": "somebody",
      }
    } />
  <div
    className="issue-detail__comment__body">
    <ReactMarkdown
      className="markdown"
      containerTagName="div"
      source="a comment" />
  </div>
</div>
`;
